modules/test/playwright/pages/commerce/commerceCatalogPage.ts

```diff
@@ -0,0 +1,64 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import {Locator, Page} from '@playwright/test';
+
+export class CommerceCatalogPage {
+	readonly cleanGlobalSearchBarButton: Locator;
+	readonly globalSearchBarCatalogPageButton: Locator;
+	readonly globalSearchBarCatalogPage: Locator;
+	readonly globalSearchResults: Locator;
+	readonly productDetailProductName: (
+		productName: string
+	) => Promise<Locator>;
+	readonly page: Page;
+
+	constructor(page: Page) {
+		this.cleanGlobalSearchBarButton = page.getByLabel('Clear Search');
+
+		this.globalSearchBarCatalogPageButton = page.locator(
+			'.commerce-topbar-button'
+		);
+		this.globalSearchBarCatalogPage = page
+			.locator('#search-bar')
+			.getByPlaceholder('Search');
+
+		this.globalSearchResults = page.locator(
+			`xpath=//div[@class="commerce-suggestions"]//div[@class="commerce-item commerce-item--search"]`
+		);
+
+		this.productDetailProductName = async (productName) => {
+			return page.getByRole('heading', {name: productName});
+		};
+
+		this.page = page;
+	}
+
+	async globalSearch(query: string) {
+		await this.globalSearchBarCatalogPage.waitFor({state: 'visible'});
+
+		await this.globalSearchBarCatalogPage.fill(query);
+	}
+
+	async findInSearchResult(text: string): Promise<Locator> {
```

Comment:
Move in the Locators:

`readonly globalSearchCommerceItemRow: (text: string) => Promise<Locator>;`

try to check if you can use `getByRole`